shared/js/smart-screen/card_manager.js

```diff
@@ -477,21 +478,40 @@
     updateCard: function cm_updateCard(item, index) {
       var that = this;
       this._asyncSemaphore.wait(function() {
-        if (typeof index === 'undefined') {
-          index = this._cardList.findIndex(function(elem) {
-            return elem.cardId === item.cardId;
-          });
-        }
-        if (index >= 0) {
-          // Most of the time, 'item' is directly reference to card in _cardList
-          // and by the time we reach here, the card ('item') is already
-          // updated. So don't bother to update it again.
-          if (this._cardList[index] !== item) {
-            this._cardList[index] = item;
+        var folder = this.findContainingFolder({ cardId: item.cardId });
+        if (folder) { // The card is under a folder not in _cardList
+          var list = folder.getCardList();
+          if (typeof index === 'undefined') {
+            index = list.findIndex(function(elem) {
+              return elem.cardId === item.cardId;
+            });
+          }
+          if (index >= 0) {
+            var onChanged = () => {
+              that.off('folder-changed', onChanged);
+              that.fire('card-updated', list[index], index);
```

Comment:
Can you try not to fire additional message here, but bind folder's `card-updated` to `home.onCardUpdated`? you may need to do this at to places. Just refer to the similar listener of `card-inserted` and `card-updated`.
https://github.com/mozilla-b2g/gaia/blob/master/tv_apps/smart-home/js/home.js#L169
https://github.com/mozilla-b2g/gaia/blob/master/tv_apps/smart-home/js/home.js#L270